File: 67-add-binary/add-binary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        i = len(a) - 1  
        j = len(b) - 1  
        ret = 0       
        result = []     

      
        while i >= 0 or j >= 0 or ret > 0:
            bitA = int(a[i]) if i >= 0 else 0
            bitB = int(b[j]) if j >= 0 else 0

            total = bitA + bitB + ret  
            result.append(str(total % 2))  
            ret = total // 2 

            i -= 1
            j -= 1

        return ''.join(reversed(result))

File: 67-add-binary/add-binary.ts
function addBinary(a: string, b: string): string {
    let i:number = a.length - 1
    let j:number = b.length - 1
    let ret:number = 0
    let sum:string = ""
    while(i>=0 || j>=0 || ret >0 ){
        const bita = i>=0 ? Number(a[i]) : 0
        const bitb = j>=0 ? Number(b[j]) : 0
        const res = bita + bitb + ret
        sum = (res%2).toString() + sum
        ret = Math.floor(res / 2)
        i--
        j--
    }
    return sum
};